### PR TITLE
Pass a Site into `make_request`

### PR DESCRIPTION
**Pull request by richvdh**
I want to try and kill off the odd `make_request`/`render` pairs that litter the test code: it's not reflective of how things work in practice and it's not particularly easy to work with.

This is step 1: make sure we have a Site upfront, and attach it to the fake http channel by passing it into `make_request`. Later that Site will be used when rendering the request, but for now that's still a separate step.

The commits in this PR should be reviewable independently.

### CODE REVIEW

File: tests/server.py [hunk 1]
```diff
@@ -145,6 +158,8 @@ def make_request(
     content, and return the Request and the Channel underneath.
 
     Args:
+        site: The twisted Site to associate with the Channel
+
         method (bytes/unicode): The HTTP request method ("verb").
         path (bytes/unicode): The HTTP path, suitably URL encoded (e.g.
         escaped UTF-8 & spaces and such).
```
Comment (anoadragon453) [inline tests/server.py:165]: Accidental space?

**Review by anoadragon453 — APPROVED** [review]
lgtm. Is the goal to eventually remove the `make_request` in HomeserverTestCase?

File: tests/server.py (hunk 1 above)
Comment (richvdh) [reply tests/server.py:165]: not really. Ideally the existing a params would have spaces between them too; I've just not gone through to fix them.

**Comment by richvdh** [comment]
> lgtm. Is the goal to eventually remove the make_request in HomeserverTestCase?

Probably not. I experimented with it, but ultmately the `self.reactor, self.site` boilerplater is a bit boilerplatey so nice to avoid repeating.

**Pull request merged**